seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/snowflake/SnowflakeTypeMapper.java

```diff
@@ -0,0 +1,138 @@
+    private static final String SNOWFLAKE_CHAR = "CHAR";
+    private static final String SNOWFLAKE_CHARACTER = "CHARACTER";
+    private static final String SNOWFLAKE_STRING = "STRING";
+    private static final String SNOWFLAKE_TEXT = "TEXT";
+    private static final String SNOWFLAKE_BINARY = "BINARY";
+    private static final String SNOWFLAKE_VARBINARY = "VARBINARY";
+
+    private static final String SNOWFLAKE_BOOLEAN = "BOOLEAN";
+
+    private static final String SNOWFLAKE_DATE = "DATE";
+    private static final String SNOWFLAKE_DATE_TIME = "DATE_TIME";
+    private static final String SNOWFLAKE_TIME = "TIME";
+    private static final String SNOWFLAKE_TIMESTAMP = "TIMESTAMP";
+    private static final String SNOWFLAKE_TIMESTAMP_LTZ = "TIMESTAMPLTZ";
+    private static final String SNOWFLAKE_TIMESTAMP_NTZ = "TIMESTAMPNTZ";
+    private static final String SNOWFLAKE_TIMESTAMP_TZ = "TIMESTAMPTZ";
+
+    private static final String SNOWFLAKE_GEOGRAPHY = "GEOGRAPHY";
+    private static final String SNOWFLAKE_GEOMETRY = "GEOMETRY";
+
+    private static final String SNOWFLAKE_VARIANT = "VARIANT";
+    private static final String SNOWFLAKE_OBJECT = "OBJECT";
+
+    @Override
+    public SeaTunnelDataType<?> mapping(ResultSetMetaData metadata, int colIndex)
+            throws SQLException {
+        String snowflakeType = metadata.getColumnTypeName(colIndex).toUpperCase();
+        int precision = metadata.getPrecision(colIndex);
+        int scale = metadata.getScale(colIndex);
+        switch (snowflakeType) {
+            case SNOWFLAKE_SMALLINT:
+            case SNOWFLAKE_TINYINT:
+            case SNOWFLAKE_BYTEINT:
+                return BasicType.SHORT_TYPE;
+            case SNOWFLAKE_INTEGER:
+            case SNOWFLAKE_INT:
+                return BasicType.INT_TYPE;
+            case SNOWFLAKE_BIGINT:
+                return BasicType.LONG_TYPE;
+            case SNOWFLAKE_DECIMAL:
+            case SNOWFLAKE_NUMERIC:
+            case SNOWFLAKE_NUMBER:
+                return new DecimalType(precision, scale);
+            case SNOWFLAKE_REAL:
+            case SNOWFLAKE_FLOAT4:
+                return BasicType.FLOAT_TYPE;
+            case SNOWFLAKE_DOUBLE:
+            case SNOWFLAKE_DOUBLE_PRECISION:
+            case SNOWFLAKE_FLOAT8:
+            case SNOWFLAKE_FLOAT:
+                return BasicType.DOUBLE_TYPE;
+            case SNOWFLAKE_BOOLEAN:
+                return BasicType.BOOLEAN_TYPE;
+            case SNOWFLAKE_CHAR:
+            case SNOWFLAKE_CHARACTER:
+            case SNOWFLAKE_VARCHAR:
+            case SNOWFLAKE_STRING:
+            case SNOWFLAKE_TEXT:
+            case SNOWFLAKE_VARIANT:
+            case SNOWFLAKE_OBJECT:
+                return BasicType.STRING_TYPE;
+            case SNOWFLAKE_BINARY:
+            case SNOWFLAKE_VARBINARY:
+            case SNOWFLAKE_GEOGRAPHY:
+            case SNOWFLAKE_GEOMETRY:
+                return PrimitiveByteArrayType.INSTANCE;
```

Comment:
Can you mapper SNOWFLAKE_GEOMETRY to String?

Reply:
yep, done